File: src/main/java/org/usfirst/frc/team4099/lib/util/GearVision.java
package org.usfirst.frc.team4099.lib.util;

/**
 * Created by plato2000 on 3/4/17.
 */
public class GearVision {
    private Rotation2D turnToGear;
    private double distanceToGear;

    public GearVision(double angleToGear, double distanceToGear) {
        turnToGear = Rotation2D.fromDegrees(angleToGear);
        distanceToGear = distanceToGear;
    }

    public Rotation2D getTurnAngle() {
        return new Rotation2D(turnToGear);
    }

    public double getDistance() {
        return distanceToGear;
    }

}

File: src/main/java/org/usfirst/frc/team4099/lib/util/GearVision.kt
package org.usfirst.frc.team4099.lib.util

/**
 * Created by plato2000 on 3/4/17.
 */
class GearVision(angleToGear: Double, distanceToGear: Double) {
    private val turnToGear: Rotation2D
    val distance: Double = 0.toDouble()

    init {
        var distanceToGear = distanceToGear
        turnToGear = Rotation2D.fromDegrees(angleToGear)
        distanceToGear = distanceToGear
    }

    val turnAngle: Rotation2D
        get() = Rotation2D(turnToGear)

}
